platform/linux-generic/pktio/dpdk.c

```diff
@@ -301,6 +301,40 @@ static struct rte_mempool_ops ops_stack = {
 
 MEMPOOL_REGISTER_OPS(ops_stack);
 
+#define HAS_L4_PROTO(m, proto) ((m->packet_type & RTE_PTYPE_L4_MASK) == proto)
+
+static inline int pkt_set_ol_rx(odp_pktin_config_opt_t *pktin_cfg,
+				odp_packet_hdr_t *pkt_hdr,
+				struct rte_mbuf *mbuf)
+{
+	if (pktin_cfg->bit.ipv4_chksum &&
+	    (mbuf->packet_type & RTE_PTYPE_L3_IPV4) &&
+	    mbuf->ol_flags & PKT_RX_IP_CKSUM_BAD) {
+		if (pktin_cfg->bit.drop_ipv4_err)
```

Comment:
Based on the DPDK test-pmd code it seems that checking packet_type is not actually needed. Have you tried this?

Checking PKT_RX_IP_CKSUM_BAD alone is deprecated and one should use PKT_RX_IP_CKSUM_MASK.
 E.g. (ol_flags & PKT_RX_IP_CKSUM_MASK) == PKT_RX_IP_CKSUM_BAD